assets/js/components/notifications/FirstPartyModeSetupBanner.test.js

```diff
@@ -0,0 +1,248 @@
+			isScriptAccessEnabled: true,
+		} );
+
+		registry
+			.dispatch( CORE_NOTIFICATIONS )
+			.registerNotification(
+				FPM_SETUP_BANNER_NOTIFICATION,
+				notification
+			);
+
+		registry.dispatch( CORE_USER ).receiveGetDismissedItems( [] );
+	} );
+
+	describe( 'checkRequirements', () => {
+		it( 'is active', async () => {
+			const isActive = await notification.checkRequirements(
+				registry,
+				VIEW_CONTEXT_MAIN_DASHBOARD
+			);
+
+			expect( isActive ).toBe( true );
+		} );
+
+		it( 'is not active when FPM is enabled', async () => {
+			registry.dispatch( CORE_SITE ).receiveGetFirstPartyModeSettings( {
+				isEnabled: true,
+				isFPMHealthy: true,
+				isScriptAccessEnabled: true,
+			} );
+
+			const isActive = await notification.checkRequirements(
+				registry,
+				VIEW_CONTEXT_MAIN_DASHBOARD
+			);
+
+			expect( isActive ).toBe( false );
+		} );
+
+		it( 'is not active when FPM is not healthy', async () => {
+			registry.dispatch( CORE_SITE ).receiveGetFirstPartyModeSettings( {
+				isEnabled: false,
+				isFPMHealthy: false,
+				isScriptAccessEnabled: true,
+			} );
+
+			const isActive = await notification.checkRequirements(
+				registry,
+				VIEW_CONTEXT_MAIN_DASHBOARD
+			);
+
+			expect( isActive ).toBe( false );
+		} );
+
+		it( 'is not active when script access is not enabled', async () => {
+			registry.dispatch( CORE_SITE ).receiveGetFirstPartyModeSettings( {
+				isEnabled: false,
+				isFPMHealthy: true,
+				isScriptAccessEnabled: false,
+			} );
+
+			const isActive = await notification.checkRequirements(
+				registry,
+				VIEW_CONTEXT_MAIN_DASHBOARD
+			);
+
+			expect( isActive ).toBe( false );
+		} );
+	} );
+
+	describe( 'Rendering', () => {
+		let FPMBannerComponent;
+
+		beforeAll( () => {
+			enabledFeatures.add( 'firstPartyMode' );
+
+			FPMBannerComponent = withNotificationComponentProps(
+				FPM_SETUP_BANNER_NOTIFICATION
+			)( FirstPartyModeSetupBanner );
+		} );
+
+		it( 'should render the banner', async () => {
+			const { getByRole, getByText, waitForRegistry } = render(
+				<FPMBannerComponent />,
+				{
+					registry,
+					viewContext: VIEW_CONTEXT_MAIN_DASHBOARD,
+				}
+			);
+
+			await waitForRegistry();
+
+			expect(
+				getByText(
+					'Get more comprehensive stats by collecting metrics via your own site'
+				)
+			).toBeInTheDocument();
+
+			expect(
+				getByRole( 'button', { name: 'Enable First-party mode' } )
+			).toBeInTheDocument();
+
+			expect(
+				getByRole( 'button', { name: 'Maybe later' } )
+			).toBeInTheDocument();
+		} );
+
+		it( 'should not render the banner if dismissed', () => {
+			registry
+				.dispatch( CORE_USER )
+				.receiveGetDismissedItems( [ FPM_SETUP_BANNER_NOTIFICATION ] );
+
+			const { container } = render( <FPMBannerComponent />, {
+				registry,
+				viewContext: VIEW_CONTEXT_MAIN_DASHBOARD,
+			} );
+
+			expect( container ).toBeEmptyDOMElement();
+		} );
+
+		describe( 'onCTAClick', () => {
+			const FPMSettingsEndpoint = new RegExp(
```

Comment:
Regular constants shouldn't start with a capital:

```suggestion
			const fpmSettingsEndpoint = new RegExp(
```